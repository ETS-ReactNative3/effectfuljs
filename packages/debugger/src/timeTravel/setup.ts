import config from "../config";
config.timeTravel = true;
config.mutationObserver = true;
