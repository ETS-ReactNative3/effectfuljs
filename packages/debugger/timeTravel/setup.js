"use strict";

var _config = require("../config");

_config.default.timeTravel = true;
_config.default.mutationObserver = true;