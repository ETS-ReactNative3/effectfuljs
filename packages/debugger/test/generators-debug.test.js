require("./setup");
const trace = require("./run");
trace.silent = true;
const D = require("../main");

test("generator functions tracing", function() {
  D.context.debug = true;
  global.console = { log: jest.fn(), error: console.error };
  const mod = trace(D.force(require("./__fixtures__/generators")));
  console.log("> M:", mod);
  trace(mod());
  expect(console.log.mock.calls).toMatchSnapshot();
});

test("async functions tracing", async function() {
  D.context.debug = true;
  global.console = { log: jest.fn(), error: console.error };
  const mod = trace(D.force(require("./__fixtures__/async")));
  console.log("> M:", mod);
  console.log("R", await trace.async(mod()));
  expect(console.log.mock.calls).toMatchSnapshot();
});

test("async generator functions tracing", async function() {
  D.context.debug = true;
  global.console = { log: jest.fn(), error: console.error };
  const mod = trace(D.force(require("./__fixtures__/asyncGenerators")));
  console.log("> M:", mod);
  console.log("R", await trace.async(mod()));
  expect(console.log.mock.calls).toMatchSnapshot();
});
