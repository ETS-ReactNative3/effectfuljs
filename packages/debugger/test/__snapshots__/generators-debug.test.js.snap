// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async functions tracing 1`] = `
Array [
  Array [
    "> M:",
    [Function],
  ],
  Array [
    "ENTER testBlockPar",
  ],
  Array [
    "S1:1",
  ],
  Array [
    "T1",
    Promise {
      "_U": 0,
      "_V": 1,
      "_W": 1001,
      "_X": null,
    },
  ],
  Array [
    "S1:2",
  ],
  Array [
    "S2:1",
  ],
  Array [
    "S2:2",
  ],
  Array [
    "S3:1",
  ],
  Array [
    "S3:2",
  ],
  Array [
    "T2",
    1002,
  ],
  Array [
    "S1:3",
  ],
  Array [
    "S2:3",
  ],
  Array [
    "S3:3",
  ],
  Array [
    "T3",
    1003,
  ],
  Array [
    "T1A",
    1001,
  ],
  Array [
    "OK1",
  ],
  Array [
    "OK2",
    [Error: 1],
  ],
  Array [
    "EXIT testBlockPar",
  ],
  Array [
    "R",
    1000,
  ],
]
`;

exports[`async generator functions tracing 1`] = `
Array [
  Array [
    "> M:",
    [Function],
  ],
  Array [
    "Async Generators>",
  ],
  Array [
    "Async Generators Item:",
    "> async generators level 2",
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "Async Generators Item:",
    2,
  ],
  Array [
    "simple async generator result:",
    4,
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "Async Generators Item:",
    2,
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:0",
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:1",
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:2",
  ],
  Array [
    "level 2 result:",
    "generator result: 4",
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "Async Generators Item:",
    2,
  ],
  Array [
    "simple async generator result:",
    4,
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "Async Generators Item:",
    2,
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:0",
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:1",
  ],
  Array [
    "Async Generators Item:",
    "simple async generator item:2",
  ],
  Array [
    "Async Generators Item:",
    "< async generators level 2",
  ],
  Array [
    "yield* redirect",
    "async redirect 2",
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "Async Generators Item:",
    2,
  ],
  Array [
    "yield* redirect to sync",
    4,
  ],
  Array [
    "get sync iterator",
  ],
  Array [
    "custom iterator next:",
    0,
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "custom iterator next:",
    1,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "custom iterator next:",
    2,
  ],
  Array [
    "custom async generator",
    2,
  ],
  Array [
    "get async iterator",
  ],
  Array [
    "custom async iterator next:",
    0,
  ],
  Array [
    "Async Generators Item:",
    0,
  ],
  Array [
    "custom async iterator next:",
    1,
  ],
  Array [
    "Async Generators Item:",
    1,
  ],
  Array [
    "custom async iterator next:",
    2,
  ],
  Array [
    "custom generator",
    2,
  ],
  Array [
    "yield* empty async generator:",
    "empty",
  ],
  Array [
    "yield* empty generator:",
    "empty",
  ],
  Array [
    "Async Generators Result:",
    "variousYieldStarsInAsyncGenerator",
  ],
  Array [
    "R",
    "Async Generators Resulting Value",
  ],
]
`;

exports[`generator functions tracing 1`] = `
Array [
  Array [
    "> M:",
    [Function],
  ],
  Array [
    "G6",
    "->",
  ],
  Array [
    "G6",
    0,
  ],
  Array [
    "G6",
    1,
  ],
  Array [
    "G6",
    2,
  ],
  Array [
    "G6",
    "G1",
  ],
  Array [
    "G6",
    "A:0",
  ],
  Array [
    "G6",
    "A:1",
  ],
  Array [
    "G6",
    "A:2",
  ],
  Array [
    "G6",
    "<-",
  ],
  Array [
    "G6",
    "->",
  ],
  Array [
    "G6",
    0,
  ],
  Array [
    "G6",
    1,
  ],
  Array [
    "G6",
    2,
  ],
  Array [
    "G6",
    "G1",
  ],
  Array [
    "G6",
    "A:0",
  ],
  Array [
    "G6",
    "A:1",
  ],
  Array [
    "G6",
    "A:2",
  ],
  Array [
    "G6",
    "<-",
  ],
]
`;
